presto-main/src/main/java/com/facebook/presto/operator/AggregationOperator.java

```diff
@@ -206,9 +235,27 @@ public Page getOutput()
         private final Accumulator aggregation;
         private final Step step;
         private final int intermediateChannel;
+        private final List<Integer> orderByChannels;
+        private final List<SortOrder> orderings;
+        private final int[] orderByOutputChannels;
+        private final Optional<PagesIndex> pageIndex;
+        private final PageBuilder pageBuilder;
 
         private Aggregator(AccumulatorFactory accumulatorFactory, Step step)
         {
+            this(accumulatorFactory, step, ImmutableList.of(), ImmutableList.of(), ImmutableList.of(), Optional.empty());
+        }
+
+        private Aggregator(
+                AccumulatorFactory accumulatorFactory,
+                Step step,
+                List<Type> sourceTypes,
+                List<Integer> orderByChannels,
+                List<SortOrder> orderings,
+                Optional<PagesIndex.Factory> pageIndexFactory)
+        {
+            checkArgument(step != Step.INTERMEDIATE, "intermediate aggregation not supported");
```

Comment:
what about Partial aggregation? Shouldn't this check be similar to:
```
checkArgument(step == Step.SINGLE, "Order by over distributed aggregation is not supported")
```
as the one below?